python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -225,23 +276,72 @@ def _match_file(
     return index.query(s_type.hash_from_file(path))
 
 
+def _group_hashes_by_prefix(
+    path: pathlib.Path,
+    settings: CLISettings,
+    hashes_grouped_by_prefix: t.Dict[t.Optional[t.Type[SignalType]], t.Set[str]],
+) -> None:
+    for line in path.read_text().splitlines():
+        line = line.strip()
+        if not line:
+            continue
+        components = line.split()
+        signal_type = None
+        if len(components) > 1:
+            # Assume it has a prefix
+            possible_type = components[0]
+            hash = components[1].strip()
+            try:
+                signal_type = settings.get_signal_type(possible_type)
+                hash = signal_type.validate_signal_str(hash)
```

Comment:
blocking: I don't think we are guaranteed that validate_signal_str won't throw a key error for some reason (we just say "throws an exception" in the interface). Separate the two catch blocks to prevent a collision.